R/SEIRDVAge.R

```diff
@@ -0,0 +1,506 @@
+    }
+    if(!is.numeric(times)){
+      stop('Evaluation times of the model storage format must numeric.')
+    }
+    if(!(all(diff(times) > 0))){
+      stop('Evaluation times of the model storage format must be increasing')
+    }
+    if (is.null(unlist(object@transmission_parameters)))
+      stop("Transmission parameters must be set before running.")
+    if (is.null(unlist(object@initial_conditions)))
+      stop("Initial conditions must be set before running.")
+    if (is.null(unlist(object@intervention_parameters)))
+      stop("Intervention parameters must be set before running.")
+    
+    #fetch number of age categories
+    n_age <- object@n_age_categories
+    
+    # set initial state vector
+    state <- c(S = initial_conditions(object)$S0,
+               E = initial_conditions(object)$E0,
+               I = initial_conditions(object)$I0,
+               R = initial_conditions(object)$R0,
+               V = initial_conditions(object)$V0,
+               VR = initial_conditions(object)$VR0,
+               D = initial_conditions(object)$D0,
+               cc = rep(0, n_age))
+    
+    # set parameters vector
+    parameters <- list(b = transmission_parameters(object)$beta,
+                    k = transmission_parameters(object)$kappa,
+                    g = transmission_parameters(object)$gamma,
+                    m = transmission_parameters(object)$mu,
+                    n = transmission_parameters(object)$nu,
+                    d_v = transmission_parameters(object)$delta_V,
+                    d_r = transmission_parameters(object)$delta_R,
+                    d_vr = transmission_parameters(object)$delta_VR)
+    
+    # fetch contact matrix of the instance
+    C = object@contact_matrix
+    
+    # set intervention parameters vector
+    
+    # use tstep=0.1 and tanh_slope=1 for good nice step-function-like shape of
+    # the intervention wave
+    sim_parms <- SimulationParameters(start =0, stop = tail(times, n=1),
+                                      tstep = 0.1)
+    
+    if(is.numeric(intervention_parameters(object)$coverages)){
+      int_parms <- 
+        InterventionParameters(
+          start=intervention_parameters(object)$starts,
+          stop=intervention_parameters(object)$stops,
+          coverage= intervention_parameters(object)$coverages)
+      inter_prot <- intervention_protocol(int_parms, sim_parms, 1) %>%
+        filter(time %in% times) %>%
+        .$coverage
+    }
+    else{
+      i = 1
+      inter_prot <- vector(mode = "list", length = n_age)
+      for(age_cov in intervention_parameters(object)$coverages){
+        int_parms <- 
+          InterventionParameters(
+            start=intervention_parameters(object)$starts,
+            stop=intervention_parameters(object)$stops,
+            coverage= age_cov)
+        inter_prot[[i]] <- intervention_protocol(int_parms, sim_parms, 1) %>%
+          filter(time %in% times) %>%
+          .$coverage
```

Comment:
Similarly, I think we can do away with the filter and just send inter_prot.time and inter_prot.coverage to approxfun